projects/igniteui-angular/src/lib/services/overlay/position/elastic-position-strategy.ts

```diff
@@ -1,54 +1,56 @@
-import { IPositionStrategy } from './IPositionStrategy';
-import { BaseFitPositionStrategy } from './base-fit-position-strategy';
-import { Size, HorizontalAlignment, VerticalAlignment, PositionSettings } from '../utilities';
+import { BaseFitPositionStrategy, ConnectedFit } from './base-fit-position-strategy';
+import { HorizontalAlignment, VerticalAlignment, PositionSettings } from '../utilities';
 
 /**
  * Positions the element as in **Connected** positioning strategy and resize the element
  * to fit in the view port in case the element is partially getting out of view
  */
 export class ElasticPositionStrategy extends BaseFitPositionStrategy {
-    protected fitHorizontal(element: HTMLElement, settings: PositionSettings, innerRect: ClientRect, outerRect: ClientRect) {
+    /** @inheritdoc */
+    protected fitInViewport(element: HTMLElement, settings: PositionSettings, connectedFit: ConnectedFit) {
         element.classList.add('igx-overlay__content--elastic');
-        let extend = 0;
-        switch (settings.horizontalDirection) {
-            case HorizontalAlignment.Left: {
-                extend = outerRect.left - innerRect.left;
-                if (extend > innerRect.width - this.settings.minSize.width) {
-                    extend = innerRect.width - this.settings.minSize.width;
-                }
-                break;
-            }
-            case HorizontalAlignment.Right: {
-                extend = innerRect.right - outerRect.right;
-                if (extend > innerRect.width - this.settings.minSize.width) {
-                    extend = innerRect.width - this.settings.minSize.width;
+        let transformString = '';
+        if (!connectedFit.fitHorizontal) {
+            const minReduction = Math.max(0, connectedFit.contentElementRect.width - this.settings.minSize.width);
+            const leftExtend = Math.max(0, connectedFit.viewPortRect.left - connectedFit.leftBorder);
+            const rightExtend = Math.max(0, connectedFit.rightBorder - connectedFit.viewPortRect.right);
+            const reduction = Math.min(minReduction, leftExtend + rightExtend);
+            element.style.width = `${connectedFit.contentElementRect.width - reduction}px`;
+
+            //  if direction is center and element goes off the screen in left direction we should push the
+            //  element to the right. Otherwise only the right border will move to the left which is not ok
+            if (settings.horizontalDirection === HorizontalAlignment.Center) {
+                //  the amount of translation depends on whether element goes off the screen to the left,
+                //  to the right or in both directions, as well as how much it goes of the screen and finally
+                //  on the minSize. The translation should be proportional between left and right extend
+                //  taken from the reduction
+                const translation = leftExtend * reduction / (leftExtend + rightExtend);
```

Comment:
Wouldn't this be division by zero if the element does fit?

Reply:
nope, if connectedFit.fitHorizontal === false either leftExtend or rightExtend should be more than 0